python/pipeline/utils/clocktools.py

```diff
@@ -0,0 +1,996 @@
+import numbers
+import numpy as np
+import datajoint as dj
+from scipy import interpolate
+from stimulus import stimulus
+from pipeline.exceptions import PipelineException
+from pipeline import treadmill, fuse, shared, odor
+
+pupil = dj.create_virtual_module("pipeline_eye", "pipeline_eye")
+
+
+def find_idx_boundaries(indices, drop_single_idx=False):
```

Comment:
I think there is a way to do this with ittertools.groupby, would save on a lot of code and documentation. [Here ](https://docs.python.org/3/library/itertools.html#itertools.groupby)is the documentation